catalog/service/impl/src/main/java/org/projectnessie/catalog/service/impl/CatalogServiceImpl.java

```diff
@@ -146,7 +150,7 @@ public Stream<Supplier<CompletionStage<SnapshotResponse>>> retrieveSnapshots(
             .keys(keys)
             .getWithResponse();
 
-    IcebergStuff icebergStuff = new IcebergStuff(objectIO, persist, tasksService, executor);
+    IcebergStuff icebergStuff = icebergStuff();
```

Comment:
Unrelated: Seems you start to like this weird name ;)

Reply:
`Iceberg` and `Stuff` those two word are enough to binary-encode anything ;)